com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ConfigurationDoneRequestHandler.java

```diff
@@ -63,7 +68,20 @@ private void handleDebugEvent(DebugEvent debugEvent, IDebugSession debugSession,
         Event event = debugEvent.event;
         boolean isImportantEvent = true;
         if (event instanceof VMStartEvent) {
-            // do nothing.
+            if (context.isVmStopOnEntry()) {
+                EventRequestManager manager = debugSession.getVM().eventRequestManager();
+                ArrayList<MethodEntryRequest> legacy = new ArrayList<>(manager.methodEntryRequests());
+                manager.deleteEventRequests(legacy);
+                MethodEntryRequest request = manager.createMethodEntryRequest();
+                request.setSuspendPolicy(EventRequest.SUSPEND_EVENT_THREAD);
+                request.enable();
+            }
+        } else if (event instanceof MethodEntryEvent) {
+            if (((MethodEntryEvent) event).method().name().equals("main")) {
```

Comment:
Need check the method signature is exactly matching with 'public static void main(string[] args)'.

Reply:
when the main entry is reached, should delete the methodEntryRequest.